x/sync/firewood/syncer.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package firewood
+
+import (
+	"bytes"
+	"context"
+	"errors"
+
+	"github.com/ava-labs/firewood-go-ethhash/ffi"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/maybe"
+
+	xsync "github.com/ava-labs/avalanchego/x/sync"
+)
+
+var (
+	_ xsync.DB[*RangeProof, struct{}] = (*database)(nil)
+
+	defaultSimultaneousWorkLimit = 8
+)
+
+// database wraps a Firewood FFI database to implement the xsync.DB interface.
+type database struct {
+	db *ffi.Database
+}
+
+type Config struct {
+	SimultaneousWorkLimit int                   // optional, defaults to 8
+	Log                   logging.Logger        // optional
+	StateSyncNodes        []ids.NodeID          // optional
+	Registerer            prometheus.Registerer // optional
+}
+
+func NewSyncer(config Config, db *ffi.Database, targetRoot ids.ID, rangeProofClient *p2p.Client, changeProofClient *p2p.Client) (*xsync.Syncer[*RangeProof, struct{}], error) {
+	r := config.Registerer
+	if r == nil {
+		r = prometheus.NewRegistry()
+	}
```

Comment:
The creation of these intermediate variables makes it less clear that we're updating config defaults. Since the config is a value and not a pointer - it's fine to mutate it directly since we don't mess with the caller's state
```suggestion
    if config.Registerer != nil {
        config.Registerer = prometheus.NewRegistry()
    }
```